### ISSUE TITLE
After network disconnect, the connection never recovers.

### ISSUE THREAD

**Issue by ylu999** [awslabs/aws-crt-java#226]
Attached log
[aws-crt-java-logs.txt](https://github.com/awslabs/aws-crt-java/files/5154357/aws-crt-java-logs.txt)

App first start with a good iot connection. Then I un-plug network, wait for a while like 2 minutes, then plug network back in.
Expecting iot connection will recover, but it hang in there forever.
Looked at logs, seems `[dns] - static, resolving c1awrs03vpwtt1.credentials.iot.us-west-2.amazonaws.com, unsolicited resolve count` happened too fast and stopped at `count 16`.

I'm using AwsIotMqttConnectionBuilder with
withCleanSession(false)
withWebsockets(true)
and X509CredentialsProvider

I did subscribe the same multiple times before I un-plug network. Not sure if this is a good practice or not.

Please help. Thanks!

**Comment by DavidOgunsAWS** [comment]
This is known underlying behavior [issue](https://stackoverflow.com/questions/30360029/if-getaddrinfo-fails-once-it-fails-forever-even-after-network-is-ready) with [getaddrinfo()](https://man7.org/linux/man-pages/man3/getaddrinfo.3.html). An immediate work around is to introduce an explicit call to res_init()to get things going when this condition is detected. We are looking into making the proper adjustment within aws-c-io's host resolver.

**Comment by bretambrose** [comment]
These logs appear to end on the first 15 seconds of the unplugged state.  Can you attach logs that not only include the entire period of disconnection but also at least two minutes of time once the underlying network has been restored?

**Comment by ylu999** [comment]
> These logs appear to end on the first 15 seconds of the unplugged state. Can you attach logs that not only include the entire period of disconnection but also at least two minutes of time once the underlying network has been restored?

It hangs there forever. No further logs appeared.

**Comment by bretambrose** [comment]
This should be addressed in the v1.2.8 of the v2 Java SDK.  I am working on the internal import now.

**Comment by ylu999** [comment]
Tried again. Still not got reconnect.

Here is the log. I disconnected wifi for a while, then re-enabled wifi.
Expecting shadow will resume updating to AWS IoT. But it is not.

[Aws-crt-java-connection.log](https://github.com/awslabs/aws-crt-java/files/5236691/Aws-crt-java-connection.log)

After I plug the network cable back in. Seems like [event-loop] triggered every 100s.
But it wakes up with 0 events to process.

I'm using Aws-iot-device-sdk-java-v2.
AwsIotMqttConnectionBuilder with
withCleanSession(false)
withWebsockets(true)
and X509CredentialsProvider

Any ideas?
